reader.go

```diff
@@ -27,7 +27,7 @@ func New(r io.Reader) io.Reader {
 func (r reader) Read(p []byte) (n int, err error) {
 	n, err = r.r.Read(p)
 	for i, b := range p {
-		if b == rByte {
+		if j := i + 1; b == rByte && ((j < len(p) && p[j] != nByte) || j == len(p)) {
```

Comment:
Sadly, I think this won't work in all cases. `Read()` is allowed to return less than `len(p)` bytes, even if it's not to the end of the underlying Reader. So it can happen that a `\r` falls on the last byte of a `Read()` call, then the `\n` is the first byte of the next `Read()` call.

You can test this by creating your own `Reader` implementation to pass into this for testing which simulates this unfortunate case.

One fix is to require that this reader take a `bufio.Reader` as input rather than a `io.Reader` so that you can always call `Peek()` to get the next byte (I think that works).

Another fix could be to continue taking `io.Reader` and then test whether it can be cast in `bufio.Reader` and if not manually wrapping it.